### ISSUE TITLE
Win10/Linux SSHFS-WIN - weird permission problem (similar but not same as other permission issues here)

### ISSUE THREAD

**Issue by anttin2020** [billziss-gh/sshfs-win#211]
Hi!

Sorry for longish story but I wanted to give all info I have figured out so far :)

I have SSHFS-WIN/WinFSP getting proper permission on the remote Linux system (664 for file, 755 for dirs) but in Windows 10 I have very weird problem that I have tried to solve on my own and with great help from Sirikali author. I have read about related issues but they are still not the same so it's better to start new thread :)

Background info
-Windows 10 Pro
-username has no non-ascii chars
-username belongs to Administrators group
-fsptool gives different uids for owner and group
-remote system is RedHat Linux

Problem 1 (minor but may give clue for some guru):
When I create a new file in Windows on that mounted drive (say new/text document) and then refresh the window, I get FOUR of them (New Text Document.txt, New Text Document (2).txt etc.). :o
Same happens if I create new folder, it gets four copies.
I have read about this but any solutions I have found have not helped.

Problem 2: (major)
But worse and maybe even more weird problem (though obviously related) is that if I create a new file say in Notepad++ and then try to save it to the mounted drive, I get error message:

```
N:\new 3.txt
You don’t have permission to save in this location. 
Contact the administrator to obtain permission.

Would you like to save in the Documents folder instead?
```

When I say NO to that, the FILE IS CREATED BUT IS EMPTY. After that I can open it and edit it and save it.

This is causing big problems with e.g. VSCode Local History which tries to create history files in the current folder and because of that problem fails. And of course any program that tries to create files/folders on the mounted drive.

I use Sirikali and I have discussed with the author and he has helped me a lot but this is something even he could not figure out. I even tried exactly same sshfs command line as he used without problems and still got the same problem.

My current sshfs command is (from Sirikali debug but also tried running separately, same result):
`Command: "C:\Program Files\SSHFS-Win\\bin\sshfs.exe" -f "username@mysite.com:/home/username" "N:"  -o rw,fsname=sshfs@"username@mysite.com:/home/username",subtype=sshfs,create_file_umask=0111,create_dir_umask=0000,uid=-1,gid=-1,idmap=user,StrictHostKeyChecking=no,IdentityFile=C:/Users/username/.ssh/id_rsa,volname="username"`

And the fsptool-x64 gives:
```
C:\Program Files (x86)\WinFsp\bin>fsptool-x64 id
User=S-1-5-21-2240619032-2423407457-165706809-1002(computername\username) (uid=197610)
Owner=S-1-5-32-544(BUILTIN\Administrators) (uid=544)
Group=S-1-5-21-2240619032-2423407457-165706809-513(computername\None) (gid=197121)

C:\Program Files (x86)\WinFsp\bin>fsptool-x64 perm n:\
O:S-1-5-21-2240619032-2423407457-165706809-1002G:S-1-5-21-2240619032-2423407457-165706809-513D:P(A;;FA;;;S-1-5-21-2240619032-2423407457-165706809-1002)(A;;0x1200a8;;;S-1-5-21-2240619032-2423407457-165706809-513)(A;;0x1200a8;;;WD) (perm=197610:197121:0711)
```
I also tried to create non-admin user account and run sshfs as that but same result.

The server gives umask 0002 which to me seems suitable and as I said, the files there have correct permissions, the problem is at Windows side (but may still be caused by server settings, I am no Linux pro).

The Linux system is a webhotel that is using cPanel control system so I have very limited rights to configure it (no sudo and limited commands).

Has anyone any ideas how to fix this or at least what is causing this? :)
Or any comments?

Thank you very much!

Antti

**Comment by billziss-gh** [comment]
>```
>C:\Program Files (x86)\WinFsp\bin>fsptool-x64 id
>User=S-1-5-21-2240619032-2423407457-165706809-1002(computername\username) (uid=197610)
>Owner=S-1-5-32-544(BUILTIN\Administrators) (uid=544)
>Group=S-1-5-21-2240619032-2423407457-165706809-513(computername\None) (gid=197121)
>```

The `fsptool id` command reports that the user is `computername\username` but that the owner for new files is `BUILTIN\Administrators` (and that the group for new files is `computername\None`).

This is rather unusual and is probably the reason that you are seeing the problem you are. To resolve the problem we likely have to figure out why new objects in your account are owned by `BUILTIN\Administrators`.

Is there something special about your Windows environment?

**Comment by anttin2020** [comment]
Thank you for your reply, I hadn't paid attention to that and actually now I noticed that if I run Command Prompt, it says 'Administrator' on the title and also Notepad++ always starts with that on title as well :o
On my other computer (not running SSHFS) the id gives:
```
User=S-1-5-21-1861491748-1931008658-1969894689-1001(AS\username) (uid=197609)
Owner=S-1-5-21-1861491748-1931008658-1969894689-1001(AS\username) (uid=197609)
Group=S-1-5-21-1861491748-1931008658-1969894689-513(AS\None) (gid=197121)
```
I have no idea where that difference comes from, my account belongs to Administrators group on both computers :o

If you have any ideas what could cause this, let me know but meanwhile I will need to start figuring that out :)

Thank you very much!

Antti

**Comment by anttin2020** [comment]
I compared the user settings on my computers and both were same.
The UAC was at lowest level on both BUT for some weird reason, on this computer there was notification that the change will need restart :o
I haven't touched that setting since I installed Windows over a year ago so and I have rebooted the computer multiple times of course but for some reason unknown to me now it said that.

When I rebooted after confirming that with Ok, now everything works! :O
So the UAC must have been in some weird state and that was causing the whole mess :o

Thank you VERY much for help!

Antti
